File: Example/macOS-Objective-C/NSLabel.m
//
//  NSLabel.m
//  macOS-Objective-C
//
//  Created by Felix Herrmann on 23.10.21.
//

#import "NSLabel.h"


@implementation NSLabel


#pragma mark - Properties

- (NSString *)text {
    return [self stringValue];
}

- (void)setText:(NSString *)text {
    [self setStringValue:text];
}


#pragma mark - Initializers

- (instancetype)initWithFrame:(NSRect)frameRect {
    if(self = [super initWithFrame:frameRect]) {
        self.lineBreakMode = NSLineBreakByTruncatingTail;
        self.maximumNumberOfLines = 1;
        self.bezeled = NO;
        self.drawsBackground = NO;
        self.editable = NO;
        self.selectable = NO;
    }
    
    return self;
}

@end

File: Example/macOS/NSLabel.swift
//
//  NSLabel.swift
//  macOS
//
//  Created by Felix Herrmann on 22.10.21.
//

import AppKit

final class NSLabel: NSTextField {
    
    
    // MARK: - Properties
    
    var text: String {
        get {
            return stringValue
        }
        set {
            stringValue = newValue
        }
    }
    
    
    // MARK: - Initializers
    
    override init(frame frameRect: NSRect) {
        super.init(frame: frameRect)
        
        lineBreakMode = .byTruncatingTail
        maximumNumberOfLines = 1
        isBezeled = false
        drawsBackground = false
        isEditable = false
        isSelectable = false
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
